### PR TITLE
Ensure contact delegate helper is set during initialization.

### PR DESCRIPTION
**Pull request by charlesmchen**
PTAL @michaelkirk

### CODE REVIEW

File: Signal/src/ViewControllers/ContactsViewHelper.m [hunk 1]
```diff
@@ -26,19 +26,26 @@ @interface ContactsViewHelper ()
...
 {
     self = [super init];
     if (!self) {
         return self;
     }
 
+    OWSAssert(delegate);
+    _delegate = delegate;
+
+    self.suppressDelegateEvents = YES;
```
Comment (charlesmchen) [inline Signal/src/ViewControllers/ContactsViewHelper.m]: We don't want to fire delegate events during initialization, but we do want to consult the "should hide local number" delegate method.

Comment (michaelkirk) [inline Signal/src/ViewControllers/ContactsViewHelper.m]: Why do we want the "shouldHideLocalNumber" method, it's not obvious that we're using it during init?

And there are only two delegate methods at the moment, so when you say we don't want to fire delegate events except using the shouldHideLocalNumber, you mean we don't want to fire the other one, which is just contactsViewHelperDidUpdateContacts, right?

I imagine this makes sense to you after digging a bit, but at first glance I don't know what behavior you're trying to fix. Can you leave a code comment that outlines the rationale?

Comment (michaelkirk) [inline Signal/src/ViewControllers/ContactsViewHelper.m]: Ah, sorry, with the GH split diff I missed the call to `[self updateContacts];` which uses the shouldHideLocalNumber (via another method call).

This makes sense, but a comment would still be helpful.

Comment (michaelkirk) [inline Signal/src/ViewControllers/ContactsViewHelper.m]: maybe the variable could be called "shouldNotifyDelegateOfUpdatedContacts`

Comment (charlesmchen) [inline Signal/src/ViewControllers/ContactsViewHelper.m]: Done. PTAL @michaelkirk

**Pull request merged**